Source/ui_unix/vfsmodel.cpp

```diff
@@ -0,0 +1,96 @@
+#include "vfsmodel.h"
+#include "vfsmanagerdialog.h"
+
+VFSModel::VFSModel(QObject *parent)
+    :QAbstractTableModel(parent)
+{
+    SetupDevices();
+}
+
+int VFSModel::rowCount(const QModelIndex & /*parent*/) const
+{
+   return m_devices.size();
+}
+
+int VFSModel::columnCount(const QModelIndex & /*parent*/) const
+{
+    if (m_devices.size() > 0) {
+        return 3;
+    } else {
+        return 0;
+    }
+}
+
+QVariant VFSModel::data(const QModelIndex &index, int role) const
+{
+    if (role == Qt::DisplayRole)
+    {
+        CDevice* m_device = m_devices.at(index.row());
+        const char* val;
+        switch (index.column()){
+        case 0:
+            val = m_device->GetDeviceName();
+            break;
+        case 1:
+            val = m_device->GetBindingType();
+            break;
+        case 2:
+            val = m_device->GetBinding();
+            break;
+        }
+        return QVariant(val);
+    }
+    return QVariant();
+}
+
+void VFSModel::SetupDevices()
+{
+    m_devices[0] = new CDirectoryDevice("mc0",  "ps2.mc0.directory");
```

Comment:
Doesn't seem to be deleted anywhere. Would probably better to use `shared_ptr` or `unique_ptr` here.